dask/array/routines.py

```diff
@@ -1002,7 +1002,12 @@ def squeeze(a, axis=None):
 
 @derived_from(np)
 def compress(condition, a, axis=None):
-    condition = asarray(condition).astype(bool)
+
+    if not isinstance(condition, np.ndarray):
```

Comment:
We're starting to also support other array types, like `sparse.COO` and `cupy.ndarray`, so we try to avoid explicit type checks against `np.ndarray` if possible.  

I wonder in which cases this is necessary, perhaps when we pass in something like a list?  If so, I recommend the following:

```python
from dask.utils import is_arraylike  # up at the top of the file

if not is_arraylike(condition):
    condition = asarray(condition)
```